apps/condo/domains/user/integration/sberid/routes.js

```diff
@@ -0,0 +1,134 @@
+const { isNil } = require('lodash')
+const { generators } = require('openid-client')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getSchemaCtx } = require('@open-condo/keystone/schema')
+
+const { getOnBoardingStatus } = require('@condo/domains/organization/integrations/sbbol/sync/getOnBoadringStatus')
+const { RESIDENT, SBER_ID_SESSION_KEY } = require('@condo/domains/user/constants/common')
+const { SberIdIdentityIntegration } = require('@condo/domains/user/integration/sberid/SberIdIdentityIntegration')
+const { syncUser } = require('@condo/domains/user/integration/sberid/sync/syncUser')
+const {
+    getRedirectUrl,
+    getUserType,
+    validateState,
+    validateNonce,
+    hasSamePhone,
+} = require('@condo/domains/user/integration/sberid/utils')
+const {
+    User,
+} = require('@condo/domains/user/utils/serverSchema')
+
+// init constants
+const integration = new SberIdIdentityIntegration()
+const logger = getLogger('sberid/routes')
+
+class SberIdRoutes {
+    async startAuth (req, res, next) {
+        try {
+            const redirectUrl = getRedirectUrl(req)
+            const userType = getUserType(req)
+
+            // clean state if anything exists in the session
+            delete req.session[SBER_ID_SESSION_KEY]
+
+            // generate checks
+            const checks = { nonce: generators.nonce(), state: generators.state() }
+
+            // set session info
+            req.session[SBER_ID_SESSION_KEY] = { checks, redirectUrl, userType }
+            await req.session.save()
+
+            const authFormUrl = await integration.generateLoginFormParams(checks, redirectUrl)
+            return res.redirect(authFormUrl)
+        } catch (error) {
+            return next(error)
+        }
+    }
+
+    async completeAuth (req, res, next) {
+        const reqId = req.id
+        try {
+            const { keystone: context } = await getSchemaCtx('User')
+            const redirectUrl = getRedirectUrl(req)
+            const userType = getUserType(req)
+
+            // validate state parameter
+            validateState(req)
+
+            // getting tokenSet from user external provider
+            let tokenSet, userInfo
+            try {
+                tokenSet = await integration.issueExternalIdentityToken(req.query.code, redirectUrl)
+            } catch (err) {
+                logger.error({ msg: 'SberId completeAuth error', err, reqId })
+                throw err
+            }
```

Comment:
Does it look like the same try-catch logic?

Reply:
The same logic like in sbbol implementation